LibSWBF2/API.cpp

```diff
@@ -99,6 +100,80 @@ namespace LibSWBF2
 		modelCount = (uint32_t)modelPtrs.Size();
 	}
 
+	//TEMPORARY: Basic texture handling until I push the other wrappers...
+    const bool Level_GetTextureData(const Level* level, const char *texName, const uint8_t*& imgData, int& width, int& height)
+    {
+    	const Texture *tex = level -> GetTexture(texName);
+    	if (tex == nullptr)
+    	{
+    		return false;
+    	}
+
+    	uint16_t w,h;
+    	tex -> GetImageData(ETextureFormat::R8_G8_B8_A8, 0, w, h, imgData);
+    	height = h;
+    	width = w;
+    	return true;
+    }
```

Comment:
This is pretty raw, I'm thinking about adding different formats to the texture wrapper to hold other kinds of data (eg, <= 16 channel terrain blend map textures or a single 1-byte channel for different per vertex terrain flags (holes, water, etc)).  Until then I kept things primtive

Reply:
Holes are actually baked into the patch meshes (IBUF, VBUF). You already figured how water is stored?

Reply:
> Holes are actually baked into the patch meshes
Baked into meaning the vertices/indices are omitted, or they have special values/flags?  I assume they're omitted since their corresponding verts are zeroed in Unity...  Also, are 'holes' and 'cuts' synonymous?  If not, are 'cuts' stored in ```CUTR``` chunks? 

As regards the ```Texture``` wrapper question, I figured it would be nice to include holes/water as mask textures since Unity requests a holes texture for terrain and whatever water system I end up using would probably use one as well.   

> You already figured how water is stored?
Nah, haven't started on that yet.

Reply:
Yea I checked the same place on geonosis (lol) and noticed the terrain mesh is rendered at greater precision than the heightmap samples the physics engine (most probably) uses.  This'll be interesting to workaround in Unity, perhaps I'll just keep the heightmap system and write a simple blend shader for the layer textures and import the terrain as any another mesh.  Will adjust the heightmap code to filter out vertices occurring at non-regular positions...

Reply:
I don't use Unreals's terrain system either, mainly because it doesn't work at runtime, but also because I can simply import the vertex and index buffers as with any other mesh, and the terrain holes are right there. The terrain wrapper takes care of patching all PTCH chunks together into a single vertex/index/normal/color/uv buffer.

Reply:
Do you just use a raw mesh collider (Unreal equivalent) to handle collision with the non-uniform terrain mesh?

Reply:
I'm currently not dealing with collision at all, since collision meshes are still on the todo list. But the component I use provides collision options yes.